osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -75,6 +75,27 @@ public abstract class Mod : IMod, IEquatable<Mod>, IDeepCloneable<Mod>
             }
         }
 
+        /// <summary>
+        /// The number of settings on this mod instance which have been adjusted by the user from their default values.
+        /// </summary>
+        public int UserAdjustedSettingsCount
```

Comment:
Bit weird to have this on `Mod` given that the only usage (so far? ever?) is in `ModDifficultyAdjust`, and bit weird that `IMod.HasNonDefaultSettings` copy-pastes much of this, but I guess I don't have any real objections

Reply:
> I guess the count has less potential to be useful elsewhere so it should be preferred local?

I would go with this for now yeah. The bool flag can be global.